benchmark/BenchmarkExamples.cpp

```diff
@@ -84,9 +84,9 @@ void BM_Tables(BenchmarkRecords* records) {
   for (int row = 0; row < 2; row++) {
     for (int column = 0; column < 2; column++) {
       records->addToExistingTable("Adding exponents", row, column,
-          [&](){exponentiateNTimes(2, row+10)+exponentiateNTimes(2, column+10);});
+          [&](){size_t temp __attribute__((unused)); temp = exponentiateNTimes(2, row+10)+exponentiateNTimes(2, column+10);});
```

Comment:
Why don't you also use a random number here? 
And the unused warning indicates a really important point when benchmarking:
If we only run code (like here) but never use the results, and the compiler can prove that there are no side effects, then it may remove the whole computation.
We maybe should introduce a function `doNotOptimize` and add it to the Benchmark suite, so that you can write something like

size_t temp = randomNumber();
temp = exponentiateNTimes(temp, ...);
doNotOptimize(temp).

But then again we should maybe also include google benchmark for this functionality and use their functions, because they have the better experts for these kinds of things:)

Reply:
> Why don't you also use a random number here?
Where do you mean?

The idea of `doNotOptimize` sounds like a good one, unfortunaly I have no idea, how that function would even look like.
Additionaly, your example seems to be wrong, considering that forbidding the compiler to optimize a variable, after already doing all manipulation of it, doesn't really change the runtime of it.